web-common/src/features/dashboards/show-hide-selectors.ts

```diff
@@ -73,13 +74,20 @@ function createShowHideStore<Item>(
       };
     },
   ) as ShowHideSelectorStore;
+  const updateMethod = `updateVisible${fieldInMeta.replace(/\w/, (s) =>
+    s.toUpperCase(),
+  )}` as "updateVisibleMeasures" | "updateVisibleDimensions";
+  const localUserPreferences = getLocalUserPreferences();
```

Comment:
This feels a bit hacky. I think it stems from the `createShowHideStore(...)` function signature. Rather than...
```typescript
function createShowHideStore<Item>(
  metricsViewName: string,
  metricsView: CreateQueryResult<V1MetricsView, RpcStatus>,
  fieldInMeta: keyof Pick<V1MetricsView, "dimensions" | "measures">,
  visibleFieldInStore: keyof Pick<
    MetricsExplorerEntity,
    "visibleMeasureKeys" | "visibleDimensionKeys"
  >,
  allVisibleFieldInStore: keyof Pick<
    MetricsExplorerEntity,
    "allMeasuresVisible" | "allDimensionsVisible"
  >,
  labelSelector: (i: Item) => string,
)
```
... this could be cleaner / simpler:
```typescript
function createShowHideStore<Item>(
  metricsViewName: string,
  type: "dimensions" | "measures",
)
```